### PR TITLE
Added an example of coinduction from isect and nat

### PR DESCRIPTION
**Pull request by jozefg**
There are a few interesting proofs left that I'm not sure are possible but haven't disproved in my head yet
- `conat` is a fixed point of `conatF`.
- That all members of `conat` are `inr` of a `conat` or `inl(<>)` (remember all things are equal in `top`)
- `conat` is the greatest fixed point (this seems pretty much impossible to prove, but I can dream)

### CODE REVIEW

**Comment by jonsterling** [comment]
This is so cool! Dybjer was asking me a while back why I thought coinduction was easier to start with than induction, and this is why...

File: example/conats.jonprl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+Tactic unfolds {
+  *{unfold <conat iterate conatF top>}
+}.
+
+Tactic rauto {
+  *{reduce; auto} ||| May not terminate but shh
+}.
+
+Theorem zero-wf : [∈(inl(<>); conat)] {
```
Comment (jonsterling) [inline example/conats.jonprl]: Can we add dedicated operators for `zero` and `succ`? (or whatever names don't clash)

**Comment by jozefg** [comment]
I suppose it's a very natural consequence of intersection being much cleaner to reason about than unions as Constable was saying.

**Comment by jonsterling** [comment]
@jozefg great point! I hadn't thought of it that way...

**Comment by jozefg** [comment]
I added some operators `csucc` and `czero` and expanded `unfolds` to do the right thing with all the operators I defined.

**Comment by jonsterling** [comment]
OK, great. Thanks so much! Very exciting.

**Pull request merged**